### DISCUSSION TITLE
x-intersection throws error: Can't find variable: IntersectionObserver

### DISCUSSION THREAD

**Discussion by alexmandrikdev** [alpinejs/alpine#3566] [7. Bugs (archived)] [answered]
Greetings!

Some of our users are getting the following error from the `x-intersect` plugin: `Can't find variable: IntersectionObserver`. 

These are the only details we know about their devices:
User agent
    Mozilla/5.0 (iPhone; CPU iPhone OS 16_3_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.3 Mobile/15E148 Safari/604.1
Platform
    iOS
Device
    iPhone
Browser
    Safari
    
Safari supports IntersectionObserver on iOS from version 12.2. Based on the user agent, their Safari version is 16.3. Is this correct? Or we can't rely on the user agent to determine the browser version?

Do you have any idea why they are getting this error anyway?

Thanks!

**Comment by ekwoka** [comment, marked as answer]
There isn't any reason I think anyone here could know about why it wouldn't work.

Maybe looking for more help at https://github.com/WebKit/WebKit would be more production.

**Reply by alexmandrikdev** [reply]
@ekwoka, you are right. This isn't an Alpine.js-related issue.
